File: Aula03/Moto.py
from Veiculo import Veiculo
class Moto(Veiculo):
    def __init__(self, marca, modelo, ano, placa, cilindradas):
        super().__init__(marca, modelo, ano, placa)
        self.__cilindradas = cilindradas

    def get_cilindradas(self):
        return self.__cilindradas

    def set_cilindradas(self, cilindradas):
        self.__cilindradas = cilindradas

    def __str__(self):
        return super().__str__() + f'\nCilindradas: {self.__cilindradas}'

File: Aula11_JavaHeranca/Moto.java
public final class Moto extends Veiculo{
    private boolean partidaEletrica;


    @Override
    public String toString() {
        var partida = this.getPartidaEletrica() ? "Sim" : "Não";
        return "Veículo: " + this.getMarca()
            + " " + this.getModelo()
            + " " + this.getAno()
            + " - Part. Elétrica: " + partida;
    }

    public boolean getPartidaEletrica() {
        return partidaEletrica;
    }
    public void setPartidaEletrica(boolean partidaEletrica) {
        this.partidaEletrica = partidaEletrica;
    }
    public Moto(String marca, String modelo, int ano, String placa, boolean partidaEletrica) {
        super(marca, modelo, ano, placa);
        this.partidaEletrica = partidaEletrica;
    }

    @Override
    public double calcularImposto() {
        return 500.0;
    }
    
    
}
